### PR TITLE
Initial version of the sdx-gateway

### PR DESCRIPTION
**Pull request by JamesGardiner**
This pull request introduces the core functionality of sdx-gateway, namely subscribing to a queue, and publishing any messages received to a different queue.

It also introduces the concept of a healthcheck endpoint to the service.

### CODE REVIEW

**Review by warrenbailey — CHANGES_REQUESTED** [review]
Please update PR description to reflect the work that's been done

File: CHANGELOG.md [hunk 1]
```diff
@@ -1 +1 @@
-## Unreleased
\ No newline at end of file
+## Unreleased
```
Comment (warrenbailey) [inline CHANGELOG.md]: Initial implementation ??

File: Pipfile.lock [hunk 2]
```diff
@@ -0,0 +1,143 @@
+{
```
Comment (warrenbailey) [inline Pipfile.lock]: Nice

File: README.md [hunk 3]
```diff
@@ -40,6 +40,10 @@ sdx-gateway by default binds to port 5000 on localhost. It exposes a single `/he
...
-| Environment Variable    | Default                               | Description
-|-------------------------|---------------------------------------|----------------
-| PORT                    | `5000`                                | Port to listen on
+| Environment Variable         | Default            | Description
+|------------------------------|--------------------|----------------
+| PORT                         | `5000`             | Port to listen on
+| SDX_GATEWAY_DEFAULT_USER     | `rabbit            | RabbitMQ username
+| SDX_GATEWAY_DEFAULT_PASSWORD | `rabbit`           | RabbitMQ password
+| SDX_GATEWAY_RABBIT_HOST      | `0.0.0.0`          | Port to listen on
+| SDX_RABBIT_GATEWAY_HOST2     | `0.0.0.0`          | Port to listen on
```
Comment (warrenbailey) [inline README.md]: SDX_GATEWAY_RABBIT_HOST2

File: app/main.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        self.rabbit_status = False
+        self.app_health = False
+
+        # Async HTTP Client
+        self.async_client = AsyncHTTPClient()
+
+    def determine_rabbit_connection_status(self):
+        for url in self.rabbit_urls:
+            try:
+                logger.exception("Fetching rabbit health")
```
Comment (warrenbailey) [inline app/main.py]: logger.debug??

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,95 @@
+aiohttp==2.2.5 \
```
Comment (warrenbailey) [inline requirements.txt:1]: Is this needed if you're using a Pipfile?

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 5 above)
Comment (JamesGardiner) [reply requirements.txt:1]: Pipfile was experimental - I'll remove the pipenv related files and let you run with standard install path.

**Review by warrenbailey — DISMISSED** [review]

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**